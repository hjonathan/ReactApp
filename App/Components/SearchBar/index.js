import React, {Component} from "react";
import {View, TextInput} from "react-native";
import I18n from "../../I18n";
import * as Res from "../../Assets/resources";
import Styles from "./styles";

const type = "search";
/**
 * Class SearchBar
 * Example:
 *      import SearchBar from "../../SearchBar";
 *
 *      class SearchBar extendes Component {
 *          render(
 *              <SearchBar />
 *          );
 *      }
 * }
 */
class SearchBar extends Component {
    constructor (props) {
        super(props);
        this.state = {
            searchInput: false
        };
    }
    /**
     * Renders searchBar
     */
    render () {
        return (
            <View style={{opacity: 0.9}}>
                <TextInput
                    style={Styles.searchBox}
                    onEndEditing={this.props.onEndEdit}
                    returnKeyType={type}
                    placeholderTextColor={Res.colors.gray}
                    placeholder={I18n.t("activity_lists_search_action_hint")}
                    underlineColorAndroid="transparent"
                />
            </View>
        );
    }
}

export default SearchBar;
