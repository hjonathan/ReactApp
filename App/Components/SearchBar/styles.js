import {StyleSheet} from "react-native";

export default StyleSheet.create({
    searchBox: {
        color: "#eee",
        marginHorizontal: 4,
        paddingHorizontal: 8,
        height: 40,
        width: 200
    }
});
