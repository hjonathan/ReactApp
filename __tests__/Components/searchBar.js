import 'react-native';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import SearchBar from '../../App/Components/SearchBar';

describe('Component >>> searchBar', () => {
    let searchBar;

    beforeEach(() => {
        searchBar = shallow(<SearchBar />);
    });

    it('Render searchBar', () => {
        expect(toJson(searchBar)).toMatchSnapshot();
    });
});
