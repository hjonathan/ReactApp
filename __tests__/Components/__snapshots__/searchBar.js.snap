// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component >>> searchBar Render searchBar 1`] = `
<Component
  style={
    Object {
      "opacity": 0.9,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    placeholder="activity_lists_search_action_hint"
    placeholderTextColor="#BABABA"
    returnKeyType="search"
    style={
      Object {
        "color": "#eee",
        "height": 40,
        "marginHorizontal": 4,
        "paddingHorizontal": 8,
        "width": 200,
      }
    }
    underlineColorAndroid="transparent"
  />
</Component>
`;
